CommonMC/src/CosmicTimeOffset_module.cc

```diff
@@ -82,6 +90,23 @@ namespace mu2e {
     std::unique_ptr<SimTimeOffset> toff(new SimTimeOffset(flatTime_.fire()));
     if( verbosityLevel_ > 1) std::cout << "CosmicTimeOffset " << toff->timeOffset_ << std::endl;
     event.put(std::move(toff));
+
+    if (addTimeOffset_ ) {
+      // Add offset to all particles
+      art::Handle<mu2e::GenParticleCollection> cosmicParticles;
+      event.getByLabel(cosmicModuleLabel_, cosmicParticles);
+      const GenParticleCollection &particles(*cosmicParticles);
+      std::unique_ptr<GenParticleCollection> offsetParticles(new GenParticleCollection);
+
+      for (GenParticleCollection::const_iterator i = particles.begin(); i != particles.end(); ++i) {
+        GenParticle const &particle = *i;
+        offsetParticles->push_back(GenParticle(particle.pdgId(), particle.generatorId(),
+                                               particle.position(), particle.momentum(),
+                                               particle.time() + flatTime_.fire()));
```

Comment:
This will give a different random time offset to every particle.  You want the same random offset to all particles in the same event.

Reply:
True, fixed.